app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -276,6 +276,21 @@ public boolean onQueryTextChange(String s) {
 
             MenuItemUtils.refreshLockItem(getActivity(), menu);
 
+            // Show Lock Item only if queue is sorted manually
+            boolean sortedAutomatically = UserPreferences.isQueueSortedAutomatically();
+            MenuItem lockItem = menu.findItem(R.id.queue_lock);
+            lockItem.setVisible(!sortedAutomatically);
+
+            // Set keep sorted checkbox
+            MenuItem sortedAutomaticallyItem = menu.findItem(R.id.queue_sort_automatically);
+            sortedAutomaticallyItem.setChecked(sortedAutomatically);
+
+            // Hide sort menu items for sort orders that are not supported by automatic sort.
+            MenuItem sortRandomItem = menu.findItem(R.id.queue_sort_random);
+            sortRandomItem.setVisible(!sortedAutomatically);
+            MenuItem sortSmartShuffleItem = menu.findItem(R.id.queue_sort_smart_shuffle);
+            sortSmartShuffleItem.setVisible(!sortedAutomatically);
```

Comment:
Smart Shuffle is deterministic, as far as I remember. That item can stay enabled.

Reply:
You are right, I will also keep the Smart Shuffle.